exchange/template.hpp

```diff
@@ -0,0 +1,65 @@
+
+#pragma once
+
+#include <functional>
+#include <string>
+
+using MarketOrderFunc = std::function<bool(std::string, std::string, double, double)>;
+
+extern "C" {
+
+class Strategy {
+public:
+    explicit Strategy(MarketOrderFunc);
+
+    ~Strategy();
+
+    /**
+     * Called whenever two orders match. Could be one of your orders, or two other
+     * people's orders.
+     *
+     * @param ticker Ticker of the orders that were matched ("ETH", "BTC", or
+     * "LTC)
+     * @param side Side of the orders that were matched ("BUY" or "SELL")
+     * @param price Price that trade was executed at
+     * @quantity quantity Volume traded
+     */
+    void on_trade_update(std::string, std::string, double, double);
```

Comment:
We should just have them write a .hpp for simplicity (for them and for us - way easier to have them upload one hpp file that we can compile)

Reply:
Also allows us to #include it into our own header with the `extern "C"` block, rather then giving them it and asking them not to modify it.